drivers/usb/device/usb_dc_stm32.c

```diff
@@ -265,25 +265,22 @@ static int usb_dc_stm32_clock_enable(void)
 
 #endif /* RCC_CFGR_OTGFSPRE / RCC_CFGR_USBPRE */
 
-#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_otghs)
-#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_usbphyc)
+#if USB_OTG_HS_ULPI_PHY
 	LL_AHB1_GRP1_EnableClock(LL_AHB1_GRP1_PERIPH_OTGHSULPI);
```

Comment:
This modification will cause compilation error on the **STM32H747-Disco** Board  because it will compile a wrong condition and search for  the clock flag "**LL_AHB1_GRP1_PERIPH_OTGHSULP**"  that  was not defined  on stm32h7xx_ll_bus file  . 
To fix the problem, i think its better to modify the condition with
```c
#if USB_OTG_HS_ULPI_PHY && DT_HAS_COMPAT_STATUS_OKAY(st_stm32_usbphyc)

```

Reply:
Aren't ULPI and PHYC incompatible with each other, and thus would never be active at the same time? My understanding is that there's either an external HS PHY connected via ULPI, or (on some models) an internal HS PHY called PHYC, or there's no HS PHY at all (FS only).

Reply:
Yes you're right it shouldn't be activated together, i guess on stm32h7 it should use another flag to enable the clock **LL_AHB1_GRP1_PERIPH_USB1OTGHSULPI**

Reply:
Oh, sorry, I had misread "h7" as "f7" - yeah, h7 has a slightly different name for the define, I've fixed that now.